lib/arc/actions/store.ex

```diff
@@ -81,6 +93,13 @@ defmodule Arc.Actions.Store do
         file_name = Arc.Definition.Versioning.resolve_file_name(definition, version, {file, scope})
         file      = %Arc.File{file | file_name: file_name}
         result = definition.__storage.put(definition, version, {file, scope})
+
+        # If this version was transformed in any way, a tempfile was created
+        # that we need to clean up.
+        if file.tempfile? do
```

Comment:
It looks like this is the same code as in `cleanup/2`. I would suggest replacing this (and the ending return of `result`) with the function call `cleanup(result, file)` to minimize unnecessary code duplication.

Reply:
Good catch, thank you!